src/core/ApolloClient.ts

```diff
@@ -443,7 +443,7 @@ export class ApolloClient implements DataProxy {
     options: WatchQueryOptions<TVariables, TData>
   ): ObservableQuery<TData, TVariables> {
     if (this.defaultOptions.watchQuery) {
-      options = mergeOptions(this.defaultOptions.watchQuery, options);
+      options = mergeOptions(this.defaultOptions.watchQuery, options as any);
```

Comment:
```suggestion
      options = mergeOptions(this.defaultOptions.watchQuery as Partial<WatchQueryOptions<TVariables, TData>>, options);
```
Maybe this cast is a bit nicer?